visualizations/vis_class.py

```diff
@@ -0,0 +1,24 @@
+from collections.abc import MutableMapping
+from abc import abstractmethod
+
+
+class Visualization(MutableMapping):
+    @abstractmethod
+    def __init__(self, filename):
```

Comment:
`filename` -> `file_name` is more Pythonic. 🐍

Reply:
I feel like `file_name` is much more consistent with the rest of the application, but it's not something I'll hold up the PR for: https://peps.python.org/pep-0008/#descriptive-naming-styles